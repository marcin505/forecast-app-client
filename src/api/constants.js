export const URL = 'http://www.kurde555.pl/';