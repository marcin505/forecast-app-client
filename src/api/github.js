import request from './request.js'
const getUser = user => request(`https://api.github.com/users/${user}`)
export { getUser };