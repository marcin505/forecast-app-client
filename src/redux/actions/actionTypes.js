/* Auth */

export const LOGGIN = 'LOGGIN';
export const LOGGIN_FAILED = 'LOGGIN_FAILED';
export const LOGGIN_SUCCESS = 'LOGGIN_SUCCESS';
export const LOGOUT = 'LOGOUT';
export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';

/* Employers */
export const SET_NOMINATED_PERSON_ID = 'SET_NOMINATED_PERSON_ID';
export const ADD_NOMINATION = 'ADD_NOMINATION';
export const RAISE_VOTES = 'RAISE_VOTES';

/* Nominations */
export const SET_NOMINATION_ADDED = 'SET_NOMINATION_ADDED';

/* menu*/

export const TOGGLE_MENU = 'TOGGLE_MENU';
