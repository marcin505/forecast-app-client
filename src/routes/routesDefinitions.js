export const MAIN = '/';
export const LOGIN = '/login';

/* USER ROUTES */
export const USER = '/user/';
export const USER_HOME = '/user/home';
export const USER_HISTORY = '/user/history';
export const USER_CONTACT = '/user/contact';

/* ADMIN ROUTES */
export const ADMIN = '/admin';
export const ADMIN_HOME = '/admin/home';
export const ADMIN_WINNER = '/admin/winner';
