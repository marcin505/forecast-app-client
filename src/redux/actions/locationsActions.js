import {
  CITY_SEARCH,
  CITY_SEARCH_SUCCESS,
  CITY_SEARCH_FAILED,
} from 'redux/actions/actionTypes';
import { CITY_SEARCH_FAILED } from './actionTypes';

export const citySearch = ({ query }) => ({
    type: CITY_SEARCH,
    payload: {
      query,
    },
  });
  
export const citySearchSuccess = ({ cities }) => ({
    type: CITY_SEARCH_SUCCESS,
    payload: {
      cities,
    },
  });
  
export const citySearchFailed = () => ({
    type: CITY_SEARCH_FAILED,
});
