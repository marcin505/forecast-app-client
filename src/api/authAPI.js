import * as axios from 'axios';
import { URL } from '../routes/routesDefinitions';

export const loginRequest = (email, password) => (axios
    .post(`${URL}/login`, {email, password}, { withCredentials: true })
    .then(res => res.data)
);
export const logoutRequest = () => (axios
    .get(`${URL}/logout`)
    .then(res => res.data)
);

export const profileRequest = (token) => {

   var config = {
      headers: {'x-access-token': token},
      withCredentials: true,
   };

   return (axios
      .get(`${URL}/profile` , config)
      .then(res => res.data)
   );

};
// http://codeheaven.io/how-to-use-axios-as-your-http-client/