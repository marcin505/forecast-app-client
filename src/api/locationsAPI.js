import * as axios from 'axios';
import { weatherURL, citySearch } from '../routes/routesDefinitions';
const apikey = 'iJ2s6KXFWURBhMLcF59iSGhukAghddZz';

export const citySearchRequest = query => (axios
    .get(`${weatherURL}${citySearch}?apikey=${apikey}&q=${query}&details=true`)
    .then(res => res.data)
);

// http://dataservice.accuweather.com/locations/v1/cities/search?apikey=iJ2s6KXFWURBhMLcF59iSGhukAghddZz&q=Nashville&details=true