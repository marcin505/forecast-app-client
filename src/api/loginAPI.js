import * as axios from 'axios';
import { URL } from 'api/constants';

export const loginRequest = email => (axios
    .post(`${URL}/login/email?email=${email}`, {}, { withCredentials: true })
    .then(res => res.data)
);
