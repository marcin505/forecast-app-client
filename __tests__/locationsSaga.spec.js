import { takeEvery, delay, takeLatest, fork } from 'redux-saga'
import { expectSaga } from 'redux-saga-test-plan';
import { call, put, take } from 'redux-saga/effects'
import * as LocationsActions from '../src/redux/actions/locationsActions';
import locationsSagas, { citySearchSaga } from '../src/redux/sagas/locationsSaga';
import { CITY_SEARCH, CITY_SEARCH_FAILED } from '../src/redux/actions/actionTypes';
import * as matchers from 'redux-saga-test-plan/matchers';
import { citySearchRequest } from '../src/api/locationsAPI';
import assert from 'assert';
import { citySearchFailed } from '../src/redux/actions/locationsActions';
import { citySearch } from '../src/routes/routesDefinitions';

//*******************************************************************************************************//

describe('>>>> test locationsSagas', () => {
    it('should have the expected watchers', done => {
        expectSaga(locationsSagas)
            .run({ silenceTimeout: true })
            .then(saga => {
                expect(saga).toMatchSnapshot();
                done();
            });
    });
    it('should put a CITY_SEARCH_SUCCESS', () => expectSaga(citySearchSaga, { payload: { query: '' } })
        .provide([
            // Use the `call.fn` matcher from Redux Saga Test Plan
            [matchers.call.fn(citySearchRequest), [{ city: 'Oslo' }]],
        ])
        .put({
            type: 'CITY_SEARCH_SUCCESS',
            payload: { cities: [{ city: 'Oslo' }] },
        })
        .run()
    );
    it('should throw an error on locationsSaga', () => {
        const generator = citySearchSaga(LocationsActions.citySearch({ query: 'Oslo' }));
        const error = new Error('Failed to fetch cities');
        assert.deepEqual(
            generator.next().value,
            call(citySearchRequest, 'Oslo'),
            'waiting for fetch response'
        )
        assert.deepEqual(
            generator.throw(error).value,
            put({ type: 'CITY_SEARCH_FAILED', payload: { error } }),
            'putting failure action'
        )
    });
});


// assert.deepEqual(iterator.throw('error').value, put(actions.checkoutFailure('error')))