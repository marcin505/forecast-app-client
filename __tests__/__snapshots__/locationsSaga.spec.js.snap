// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>> test locationsSagas should have the expected watchers 1`] = `
Object {
  "fork": Array [
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [
          "CITY_SEARCH",
          "@@redux-saga-test-plan/json/function/citySearchSaga",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeLatest",
      },
    },
  ],
  "take": Array [
    Object {
      "@@redux-saga/IO": true,
      "TAKE": Object {
        "pattern": "CITY_SEARCH",
      },
    },
  ],
}
`;
